packages/suite/src/hooks/suite/useChangeDeviceLabel.ts

```diff
@@ -0,0 +1,55 @@
+import { useForm } from 'react-hook-form';
+
+import { yupResolver } from '@hookform/resolvers/yup';
+
+import { yup } from '@suite-common/validators';
+import { selectDeviceLabel, selectDeviceName } from '@suite-common/wallet-core';
+import { EventType, analytics } from '@trezor/suite-analytics';
+import { isAscii } from '@trezor/utils';
+
+import { applySettings } from 'src/actions/settings/deviceSettingsActions';
+import { MAX_LABEL_LENGTH } from 'src/constants/suite/device';
+import { useDispatch, useSelector, useTranslation } from 'src/hooks/suite';
+
+import { TranslationFunction } from './useTranslation';
+
+const changeDeviceLabelSchema = (t: TranslationFunction) =>
+    yup.object({
+        deviceLabel: yup
+            .string()
+            .max(
+                MAX_LABEL_LENGTH,
+                t('TR_LABEL_ERROR_LENGTH', {
+                    length: MAX_LABEL_LENGTH,
+                }),
+            )
+            .test({
+                test: value => isAscii(value),
```

Comment:
small nit to simplify:
```suggestion
                test: isAscii,
```